app/jobs/activity_tracking_job.rb

```diff
@@ -0,0 +1,23 @@
+class ActivityTrackingJob < ApplicationJob
+  queue_as :dradis_project
+
+  def perform(action:, trackable_id:, trackable_type:, user:)
```

Comment:
just pass the user email here?

Reply:
I would prefer to keep the user as it is. I think that in the future we are going to change user.email for user.id in most models, like in Activity, and I already have the user ready this way.
Also, we may need to pass the user to create other notifications, not sure yet.

Reply:
I don't think you can pass the `User` in directly, or any "complex" Ruby object. In production* jobs use Resque, which means they'll be serialized using Redis, which can only store primitive types like strings and integers.

I've run into this issue before on another project and had to pass the record's `id` to the job, then inside the job load it again with `Record.find(id)`. (If you want to be really pedantic, reloading the record is probably a good idea anyway because the DB row may have changed since the job was queued and the job may need the most up-to-date data... unlikely to be an issue here but worth bearing in mind for future BG jobs.)

*or at least we use Resque in all envs in Pro? See `config/application.rb` in that repo. Not sure if this is the case in CE even though `resque` is included in CE's Gemfile. Whatever the case these changes are going to merged into Pro eventually so we should make them compatible with Resque now.

This is actually a great example of why it's a good idea to use the same ActiveJob adapter in all environments for the sake of [dev-prod parity](https://12factor.net/dev-prod-parity)... otherwise these kind of issues slip through the net.

Reply:
Updated to use user_id